CMake export Half dependency

When installing plugins via pip:
```
CMake Error at /home/msalinas/miniconda3/envs/xmipp4/lib/cmake/xmipp4-core/modules/Findhalf.cmake:35 (file):
    file STRINGS file
    "/home/msalinas/Documents/standalone-installations/xmipp4/xmipp4-binding-python/src/half_INCLUDE_DIR-NOTFOUND/half.hpp"
    cannot be read.
  Call Stack (most recent call first):
    /usr/share/cmake-3.28/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
    /home/msalinas/miniconda3/envs/xmipp4/lib/cmake/xmipp4-core/xmipp4-core-config.cmake:29 (find_dependency)
    src/CMakeLists.txt:39 (find_package)


  CMake Error at /home/msalinas/miniconda3/envs/xmipp4/lib/cmake/xmipp4-core/modules/Findhalf.cmake:40 (string):
    string sub-command REGEX, mode MATCH needs at least 5 arguments total to
    command.
  Call Stack (most recent call first):
    /usr/share/cmake-3.28/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
    /home/msalinas/miniconda3/envs/xmipp4/lib/cmake/xmipp4-core/xmipp4-core-config.cmake:29 (find_dependency)
    src/CMakeLists.txt:39 (find_package)


  CMake Error at /usr/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
    Could NOT find half (missing: half_INCLUDE_DIR)
  Call Stack (most recent call first):
    /usr/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
    /home/msalinas/miniconda3/envs/xmipp4/lib/cmake/xmipp4-core/modules/Findhalf.cmake:44 (find_package_handle_standard_args)
    /usr/share/cmake-3.28/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
    /home/msalinas/miniconda3/envs/xmipp4/lib/cmake/xmipp4-core/xmipp4-core-config.cmake:29 (find_dependency)
    src/CMakeLists.txt:39 (find_package)


  -- Configuring incomplete, errors occurred!
```

--- a/cmake/config/config.cmake.in
+++ b/cmake/config/config.cmake.in
@@ -22,6 +22,8 @@
 
 @PACKAGE_INIT@
 
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/modules)
+
 include(CMakeFindDependencyMacro)
 
 find_dependency(Threads)

--- a/cmake/config/config.cmake.in
+++ b/cmake/config/config.cmake.in
@@ -20,14 +20,12 @@
 #  e-mail address 'xmipp@cnb.csic.es'
 # ***************************************************************************
 
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/modules)
+@PACKAGE_INIT@
 
 include(CMakeFindDependencyMacro)
 
-find_dependency(Threads REQUIRED)
+find_dependency(Threads)
 find_dependency(spdlog)
 find_dependency(half)
 
 include(${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-targets.cmake)
-
-@PACKAGE_INIT@

--- a/cmake/modules/Findhalf.cmake
+++ b/cmake/modules/Findhalf.cmake
@@ -32,13 +32,15 @@ find_path(
 )
 
 # Parse version from header file
-file(
-    STRINGS "${half_INCLUDE_DIR}/half.hpp"
-    half_VERSION_LINE
-    REGEX "Version"
-)
-string(REGEX MATCH "Version ([0-9]*\.[0-9]*\.[0-9]*)" _ ${half_VERSION_LINE})
-set(half_VERSION ${CMAKE_MATCH_1})
+if(half_INCLUDE_DIR)
+    file(
+        STRINGS "${half_INCLUDE_DIR}/half.hpp"
+        half_VERSION_LINE
+        REGEX "Version"
+    )
+    string(REGEX MATCH "Version ([0-9]*\.[0-9]*\.[0-9]*)" _ ${half_VERSION_LINE})
+    set(half_VERSION ${CMAKE_MATCH_1})
+endif()
 
 # Define the package variables
 find_package_handle_standard_args(half 

--- a/cmake/modules/fetch_half.cmake
+++ b/cmake/modules/fetch_half.cmake
@@ -25,46 +25,46 @@ cmake_minimum_required(VERSION 3.12)
 include(FetchContent)
 
 function(fetch_half)
-    set(options)
-    set(oneValueArgs VERSION)
-    set(multiValueArgs)
-    cmake_parse_arguments(PARSE_ARGV 0 arg
-        "${options}" "${oneValueArgs}" "${multiValueArgs}"
-    )
+	set(options)
+	set(oneValueArgs VERSION)
+	set(multiValueArgs)
+	cmake_parse_arguments(PARSE_ARGV 0 arg
+		"${options}" "${oneValueArgs}" "${multiValueArgs}"
+	)
 
 	cmake_policy(SET CMP0135 NEW) # To avoid warnings
-    set(NO_OP_COMMAND "${CMAKE_COMMAND} -E true")
-    FetchContent_Declare(
-        half
-        URL https://kumisystems.dl.sourceforge.net/project/half/half/${arg_VERSION}/half-${arg_VERSION}.zip
-        CONFIGURE_COMMAND ${NO_OP_COMMAND}
-        BUILD_COMMAND ${NO_OP_COMMAND}
-        INSTALL_COMMAND ${NO_OP_COMMAND}
-    )
-    FetchContent_MakeAvailable(half)
+	set(NO_OP_COMMAND "${CMAKE_COMMAND} -E true")
+	FetchContent_Declare(
+		half
+		URL https://kumisystems.dl.sourceforge.net/project/half/half/${arg_VERSION}/half-${arg_VERSION}.zip
+		CONFIGURE_COMMAND ${NO_OP_COMMAND}
+		BUILD_COMMAND ${NO_OP_COMMAND}
+		INSTALL_COMMAND ${NO_OP_COMMAND}
+	)
+	FetchContent_MakeAvailable(half)
 
 	set(half_INCLUDE_DIR ${half_SOURCE_DIR}/include)
 
 	# Define the target
 	add_library(half INTERFACE)
 	target_include_directories(
-        half 
-        INTERFACE
-            $<BUILD_INTERFACE:${half_INCLUDE_DIR}>
-            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
-    )
+		half 
+		INTERFACE
+			$<BUILD_INTERFACE:${half_INCLUDE_DIR}>
+			$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+	)
 
-    # Define install command
-    install(FILES ${half_INCLUDE_DIR}/half.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-    install(
-        TARGETS half
-        EXPORT half-targets
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/half
-    )
-    install(
-        EXPORT half-targets
-        FILE half-config.cmake
-        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/half
-    )
+	# Define install command
+	install(FILES ${half_INCLUDE_DIR}/half.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+	install(
+		TARGETS half
+		EXPORT half-targets
+		LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/half
+	)
+	install(
+		EXPORT half-targets
+		FILE half-config.cmake
+		DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/half
+	)
 
 endfunction()

--- a/cmake/modules/fetch_half.cmake
+++ b/cmake/modules/fetch_half.cmake
@@ -44,9 +44,27 @@ function(fetch_half)
     FetchContent_MakeAvailable(half)
 
 	set(half_INCLUDE_DIR ${half_SOURCE_DIR}/include)
-    install(FILES ${half_INCLUDE_DIR}/half.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 	# Define the target
-	add_library(half INTERFACE IMPORTED)
-	target_include_directories(half INTERFACE ${half_INCLUDE_DIR})
+	add_library(half INTERFACE)
+	target_include_directories(
+        half 
+        INTERFACE
+            $<BUILD_INTERFACE:${half_INCLUDE_DIR}>
+            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+    )
+
+    # Define install command
+    install(FILES ${half_INCLUDE_DIR}/half.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+    install(
+        TARGETS half
+        EXPORT half-targets
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/half
+    )
+    install(
+        EXPORT half-targets
+        FILE half-config.cmake
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/half
+    )
+
 endfunction()

